### PR TITLE
Alter ID sequence to bigint

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Alter ID sequence to bigint.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

I tested the sql on one of our sequences and it worked as expected.

## Complete description
As described by https://github.com/flyteorg/flyte/issues/3777, some of the ID sequences are not bigint due to historical reason. It makes sense to use bigint because integer is easy to run out of.

Note that this PR does not try to change `cycle` in order to keep backward compatibility, and also `bigint` will practically make that unnecessary.

For reference https://www.postgresql.org/docs/current/sql-altersequence.html

## Tracking Issue
Fixes https://github.com/flyteorg/flyte/issues/3777

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by kumare3** [comment]
Cc @eapolinario

**Comment by eapolinario** [comment]
@honnix , one concern with this change is the possibility of blocking the tables where sequences are not bigints already. We want to minimize downtime, esp. in the case of large tables like `task_executions` and `node_executions`. 

In the recent past we [coordinated](https://flyte-org.slack.com/archives/C03CYSL783B/p1681382011320999) tests with other members of the Spotify team. Would it be possible to run a similar test on your side? Sorry for the inconvenience, but database migrations require this extra care.

**Comment by honnix** [comment]
@eapolinario Good question. We have tested this on our production system and the change took fraction of second. It doesn't do anything to existing sequences, but only changes a single row of the sequence table.

**Comment by honnix** [comment]
It is worth noting that all the `ID` columns in different tables are all `bigint` already, at least in our production database that has been evolving through the past couple of years. It's only the sequence type being inconsistent across different sequences.

**Review by eapolinario — APPROVED** [review]
Thanks for explaining. In that case, this PR looks good.

**Comment by honnix** [comment]
Thank you! Please help merge it when suitable.
